contracts/HydraDelegation/Delegation.sol

```diff
@@ -255,9 +333,19 @@ contract Delegation is
         uint256 reward = aprCalculatorContract.applyBaseAPR(rewardIndex);
         if (reward == 0) return;
 
-        rewardWalletContract.distributeReward(delegator, reward);
+        uint256 commission = delegationCommissionPerStaker[staker];
+        if (commission == 0) {
```

Comment:
Way to improve:

    uint256 delegatorReward;
    if (commission > 0) {
        uint256 stakerCut;
        (stakerCut, delegatorReward) = _applyCommission(reward, commission);
        rewardWalletContract.distributeReward(staker, stakerCut);
        emit CommissionClaimed(staker, delegator, stakerCut);
    } else {
        delegatorReward = reward;
    }

        // Distribute reward to delegator
    rewardWalletContract.distributeReward(delegator, delegatorReward);
    emit DelegatorRewardsClaimed(staker, delegator, delegatorReward);